Don't apply 'discipline' flags to compilation run for documentation

raboof opened the pull request:
This is not needed, and in particular caused problems with `-Ywarn-unused:_` on scala 2.13.0-M5 (#26119)

raboof committed:
Avoid -Ywarn-unused:_ on Scala 2.13

helena commented on project/AkkaDisciplinePlugin.scala:
```diff
@@ -41,7 +41,17 @@ object AkkaDisciplinePlugin extends AutoPlugin with ScalafixSupport {
+            "-Yno-adapted-args",
+            // -Ywarn-unused:_ breaks 'sbt ++2.13.0-M5 akka-actor/doc'
+            // https://github.com/akka/akka/issues/26119
+            "-Ywarn-unused:_",
```
Is it all `-Ywarn-unused` or any specific ones to avoid?

helena replied:
What I mean is perhaps some still work e.g. `-Ywarn-unused:implicits` and `-Ywarn-unused:imports`. I thought I'd run this against 2.13, hrm.

raboof commented:
I'll experiment tomorrow if we can make this more narrow. Indeed compiling
still works, but (strangely) generating the docs doesn't

On Tue, 26 Feb 2019, 17:51 Helena Edelson, <notifications@github.com> wrote:

> *@helena* commented on this pull request.
> ------------------------------
>
> In project/AkkaDisciplinePlugin.scala
> <https://github.com/akka/akka/pull/26441#discussion_r260382792>:
>
> > @@ -41,7 +41,17 @@ object AkkaDisciplinePlugin extends AutoPlugin with ScalafixSupport {
>        Compile / console / scalacOptions --= Seq("-deprecation", "-Xfatal-warnings", "-Xlint", "-Ywarn-unused:imports"),
>        Compile / scalacOptions --= (CrossVersion.partialVersion(scalaVersion.value) match {
>          case Some((2, 13)) =>
> -          Seq("-Ywarn-inaccessible", "-Ywarn-infer-any", "-Ywarn-nullary-override", "-Ywarn-nullary-unit", "-Ypartial-unification", "-Yno-adapted-args")
> +          Seq(
> +            "-Ywarn-inaccessible",
> +            "-Ywarn-infer-any",
> +            "-Ywarn-nullary-override",
> +            "-Ywarn-nullary-unit",
> +            "-Ypartial-unification",
> +            "-Yno-adapted-args",
> +            // -Ywarn-unused:_ breaks 'sbt ++2.13.0-M5 akka-actor/doc'
> +            // https://github.com/akka/akka/issues/26119
> +            "-Ywarn-unused:_",
>
> What I mean is perhaps some still work e.g. -Ywarn-unused:implicits and
> -Ywarn-unused:imports. I thought I'd run this against 2.13, hrm.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/akka/akka/pull/26441#discussion_r260382792>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAIDECN1IUCVHVpLdNwAYYFdXMmK3UhVks5vRWYIgaJpZM4bR9Ar>
> .
>

helena commented:
Great, otherwise we can consider this and leave the other existing the same, so that it only affects docs?
```
Compile / doc / scalacOptions --= (CrossVersion.partialVersion(scalaVersion.value) match {
    case Some((2, 13)) => Seq("-Ywarn-unused:_")
    case _ => Nil
  }),
```

raboof committed:
Exclude warn-unused:explicits when generating docs

helena reviewed (approved):
LGTM

jrudolph commented on project/AkkaDisciplinePlugin.scala:
```diff
@@ -39,9 +39,22 @@ object AkkaDisciplinePlugin extends AutoPlugin with ScalafixSupport {
       Compile / console / scalacOptions --= Seq("-deprecation", "-Xfatal-warnings", "-Xlint", "-Ywarn-unused:imports"),
+      // -Ywarn-unused:_ breaks 'sbt ++2.13.0-M5 akka-actor/doc'
+      // https://github.com/akka/akka/issues/26119
+      Compile / doc / scalacOptions ++= (CrossVersion.partialVersion(scalaVersion.value) match {
```
Why not disable all diagnostic flags for Scaladocs in the first place?

helena replied:
👍 I should have thought of that originally, thanks for catching @jrudolph and @raboof :)

helena replied:
Changes the scope of this, perhaps we should edit the title of this issue to clarify this larger point?

raboof committed:
Don't apply discipline to the docs compilation run

This approach also means no explicit code is needed anymore to
prevent #26119 from being triggered.
